recipes/keep_awake.rb

```diff
@@ -1,80 +1,72 @@
-plist 'disable Power Nap' do
-  path '/Library/Preferences/com.apple.PowerManagement.plist'
-  entry 'DarkWakeBackgroundTasks'
-  value false
+system_preference 'disable computer sleep' do
+  preference :computersleep
+  setting 'Never'
 end
 
-plist 'enable automatic restart on power loss' do
-  path '/Library/Preferences/com.apple.PowerManagement.plist'
-  entry 'Automatic Restart On Power Loss'
-  value true
+system_preference 'disable display sleep' do
+  preference :displaysleep
+  setting 'Never'
 end
 
-plist 'set display sleep timer to zero' do
-  path '/Library/Preferences/com.apple.PowerManagement.plist'
-  entry 'Display Sleep Timer'
-  value 0
+system_preference 'disable hard disk sleep' do
+  preference :harddisksleep
+  if node['macos']['disk_sleep_disabled']
+    setting 'Never'
+  else
+    setting '10'
+  end
 end
 
-plist 'enable wake from sleep via network' do
-  path '/Library/Preferences/com.apple.PowerManagement.plist'
-  entry 'Wake On LAN'
-  value true
+system_preference 'restart if the computer becomes unresponsive' do
+  preference :restartfreeze
+  setting 'On'
 end
 
-plist 'set system sleep timer to zero' do
-  path '/Library/Preferences/com.apple.PowerManagement.plist'
-  entry 'System Sleep Timer'
-  value 0
+system_preference 'wake the computer when accessed using a network connection' do
+  preference :wakeonnetworkaccess
+  setting 'On'
+  not_if { running_in_a_vm? }
 end
 
-plist 'disable screensaver' do
-  path "/Users/#{node['macos']['admin_user']}/Library/Preferences/ByHost/com.apple.screensaver.#{hardware_uuid}.plist"
-  entry 'idleTime'
-  value 0
+system_preference 'restart after a power failure' do
+  preference :restartpowerfailure
+  setting 'On'
+  not_if { running_in_a_vm? }
 end
 
-systemsetup 'Set amount of idle time until computer sleeps to never' do
-  setting 'computersleep'
-  value 'Never'
+system_preference 'pressing power button does not sleep computer' do
+  preference :allowpowerbuttontosleepcomputer
+  setting 'Never'
+  only_if { power_button_model? }
```

Comment:
What is the reason for restricting it to certain models? This seems like something that would be included in a cookbook that wraps around this.